### PR TITLE
Using read replicas when available 

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [X]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Some parts of Mautic used to have the ability to connect to read-only replicas, which makes sense for areas where we need to make read-only queries, this feature is important because it offloads the traffic from the read-write database servers.

I refactored the code to remove deprecations and tested it on a database server with a replica.  this PR is related to #11792

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. In `local.php` add a new entry `db_host_ro => YOUR_DATABASE_SERVER`
3. Run `bin/console mautic:segments:update` 
4. You should notice queries running on th read-only replica

### CODE REVIEW

File: app/bundles/CampaignBundle/Entity/ReplicaConnectionTrait.php [hunk 1]
```diff
@@ -19,16 +19,16 @@ trait SlaveConnectionTrait
      *
      * @return Connection
      */
-    private function getSlaveConnection(ContactLimiter $limiter = null)
+    private function getReplicaConnectionTrait(ContactLimiter $limiter = null)
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/ReplicaConnectionTrait.php]: I think that the "Trait" should not be in the method name. Also, it'd be nice to have the return type.
```suggestion
    private function getReplicaConnection(ContactLimiter $limiter = null): Connection
```

**Review by escopecz — DISMISSED** [review]
The changes seem correct to me. The deprecated code was replaced with the suggested replacements. Thank you!

**Comment by escopecz** [comment]
@mabumusa1 perhaps this might be a good PR to get merged before the stable M5 release. Can you please rebase and then ask for the second tester in #t-product?

**Review by kuzmany — DISMISSED** [review]
Seems good to me

**Comment by mabumusa1** [comment]
> @mabumusa1 perhaps this might be a good PR to get merged before the stable M5 release. Can you please rebase and then ask for the second tester in #t-product?

@escopecz tomorrow will be ready

**Review by escopecz — APPROVED** [review]
Thanks! 👍

**Pull request merged**